File: src/main/java/com/github/jacekpoz/common/sendables/database/queries/user/DeleteUserQuery.java
package com.github.jacekpoz.common.sendables.database.queries.user;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.jacekpoz.common.sendables.database.queries.basequeries.UserQuery;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class DeleteUserQuery extends UserQuery {

    @JsonCreator
    public DeleteUserQuery(
            @JsonProperty("userID") long userID,
            @JsonProperty("callerID") long callerID
    ) {
        super(userID, callerID);
    }

}

File: src/main/java/com/github/jacekpoz/common/sendables/database/queries/user/DeleteUserQuery.kt
package com.github.jacekpoz.common.sendables.database.queries.user

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.jacekpoz.common.sendables.database.queries.basequeries.UserQuery

open class DeleteUserQuery @JsonCreator constructor(
    @JsonProperty("userID") userID: Long,
    @JsonProperty("callerID") callerID: Long,
) : UserQuery(userID, callerID) {

    override fun toString(): String = "DeleteUserQuery(userID=$userID, callerID=$callerID)"
}